ogr/services/gitlab/pull_request.py

```diff
@@ -118,16 +119,50 @@ def create(
         source_branch: str,
         fork_username: str = None,
     ) -> "PullRequest":
-        mr = project.gitlab_repo.mergerequests.create(
-            {
-                "source_branch": source_branch,
-                "target_branch": target_branch,
-                "title": title,
-                "description": body,
-            }
-        )
+        repo = project.gitlab_repo
+        parameters = {
+            "source_branch": source_branch,
+            "target_branch": target_branch,
+            "title": title,
+            "description": body,
+        }
+
+        if project.is_fork and (
+            fork_username is None or fork_username == project.namespace
+        ):
+            parameters["target_project_id"] = project.parent.gitlab_repo.attributes[
+                "id"
+            ]
+        elif fork_username and fork_username != project.namespace:
+            other_project = GitlabPullRequest.__get_fork(
+                fork_username,
+                project.get_forks()
+                if project.parent is None
+                else project.parent.get_forks(),
+            )
+
+            parameters["target_project_id"] = other_project.gitlab_repo.attributes["id"]
+            if project.parent is None:
+                parameters["target_project_id"] = repo.attributes["id"]
+                repo = other_project.gitlab_repo
+
+        mr = repo.mergerequests.create(parameters)
         return GitlabPullRequest(mr, project)
 
+    @staticmethod
+    def __get_fork(
+        fork_username: str, forks: List["ogr_gitlab.GitlabProject"]
+    ) -> "ogr_gitlab.GitlabProject":
+        forks = list(
+            filter(
+                lambda fork: fork.gitlab_repo.namespace["full_path"] == fork_username,
+                forks,
+            )
+        )
+        if not forks:
+            raise GitlabAPIException("Requested fork doesn't exist")
+        return forks[0]
```

Comment:
Is there a chance that we get more than one?

Reply:
It filters by username, so it shouldn't.
However after GitLab allowing to create MR from non-existing branch through API, I wouldn't be surprised :D 

More straight-forward would be to get the project directly, but that introduces problem with renamed project (either one).